### PR TITLE
R4R: refactor mint and distribution log

### PR DESCRIPTION
**Pull request by HaoyangLiu**
Target Issue: https://github.com/irisnet/irishub/issues/1051

### CODE REVIEW

File: modules/distribution/keeper/allocation.go [hunk 1]
```diff
@@ -47,42 +47,54 @@ func (k Keeper) AllocateTokens(ctx sdk.Context, percentVotes sdk.Dec, proposer s
...
 	feePool := k.GetFeePool(ctx)
 	communityPool := feePool.CommunityPool
 	allocateCoins, _ := communityPool.MulDec(percent).TruncateDecimal()
 	feePool.CommunityPool = communityPool.Minus(types.NewDecCoins(allocateCoins))
 	k.SetFeePool(ctx, feePool)
-
+	logger.Info("Spend community tax fund", "total_community_tax_fund", communityPool.ToString(), "left_community_tax_fund", feePool.CommunityPool.ToString())
 	if burn {
-		k.bankKeeper.BurnCoinsFromPool(ctx, "communityTax", allocateCoins)
+		logger.Info(fmt.Sprintf("Burn %s", allocateCoins.String()))
```
Comment (chengwenxi) [inline modules/distribution/keeper/allocation.go]: logger.Info("Burn tokens","amount":allocateCoins.String()) will better?

File: modules/distribution/keeper/allocation.go [hunk 2]
```diff
@@ -47,42 +47,54 @@ func (k Keeper) AllocateTokens(ctx sdk.Context, percentVotes sdk.Dec, proposer s
...
+		_, err := k.bankKeeper.BurnCoinsFromPool(ctx, "communityTax", allocateCoins)
+		if err != nil {
+			panic(err)
+		}
 	} else {
-		k.bankKeeper.AddCoins(ctx, destAddr, allocateCoins)
+		logger.Info(fmt.Sprintf("Send %s to account %s", allocateCoins.String(), destAddr.String()))
```
Comment (chengwenxi) [inline modules/distribution/keeper/allocation.go]: +

File: modules/distribution/types/delegator_info.go [hunk 3]
```diff
@@ -34,18 +35,20 @@ func (di DelegationDistInfo) GetDelAccum(height int64, delegatorShares sdk.Dec)
...
 //   (see comment on TakeFeePoolRewards for more info)
-func (di DelegationDistInfo) WithdrawRewards(wc WithdrawContext, vi ValidatorDistInfo,
+func (di DelegationDistInfo) WithdrawRewards(logger log.Logger, wc WithdrawContext, vi ValidatorDistInfo,
 	totalDelShares, delegatorShares sdk.Dec) (
 	DelegationDistInfo, ValidatorDistInfo, FeePool, DecCoins) {
 
 	fp := wc.FeePool
-	vi = vi.UpdateTotalDelAccum(wc.Height, totalDelShares)
+	vi = vi.UpdateTotalDelAccum(logger, wc.Height, totalDelShares)
+	logger.Debug("After updating delegation accumulation: ", "validator_distInfo", vi.String())
```
Comment (chengwenxi) [inline modules/distribution/types/delegator_info.go]: delete `:`

File: app/v0/invariants.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ func (p *ProtocolV0) assertRuntimeInvariants(ctx sdk.Context) {
 		return
 	}
 	invariants := p.runtimeInvariants()
+	ctx = ctx.WithLogger(ctx.Logger().With("module", "invariant"))
```
Comment (wukongcheng) [inline app/v0/invariants.go]: iris/invariant ?

**Pull request merged**